File: crates/source-languages/tests/fixtures/sample.cs
using System.Collections.Generic;

namespace Example.Catalog
{
    public interface IRestockable
    {
        void Restock(int units);
    }

    public class Catalog : IRestockable
    {
        public const int DefaultPageSize = 25;

        private readonly Dictionary<string, int> items = new Dictionary<string, int>();

        public int Add(string sku, int units)
        {
            var normalized = Normalize(sku);
            items.TryGetValue(normalized, out var current);
            items[normalized] = current + units;
            return current + units;
        }

        private static string Normalize(string sku)
        {
            return sku.Trim().ToUpperInvariant();
        }

        public void Restock(int units)
        {
            Add("DEFAULT", units);
        }

        public static Catalog BuildCatalog()
        {
            var catalog = new Catalog();
            catalog.Add("a-1", DefaultPageSize);
            return catalog;
        }
    }
}

File: crates/source-languages/tests/fixtures/sample.php
<?php

namespace Example\Catalog;

use ArrayObject;

const DEFAULT_PAGE_SIZE = 25;

interface Restockable
{
    public function restock(int $units): void;
}

class Catalog implements Restockable
{
    private array $items = [];

    public function add(string $sku, int $units): int
    {
        $normalized = self::normalize($sku);
        $current = $this->items[$normalized] ?? 0;
        $this->items[$normalized] = $current + $units;
        return $current + $units;
    }

    private static function normalize(string $sku): string
    {
        return strtoupper(trim($sku));
    }

    public function restock(int $units): void
    {
        $this->add('DEFAULT', $units);
    }
}

function buildCatalog(): Catalog
{
    $catalog = new Catalog();
    $catalog->add('a-1', DEFAULT_PAGE_SIZE);
    return $catalog;
}
